src/android/toga_android/window.py

```diff
@@ -78,3 +83,85 @@ def stack_trace_dialog(self, title, message, content, retry=False):
 
     def save_file_dialog(self, title, suggested_filename, file_types):
         self.interface.factory.not_implemented('Window.save_file_dialog()')
+
+    async def open_file_dialog(self, title, initial_uri, file_mime_types, multiselect):
+        """
+        Opens a file chooser dialog and returns the chosen file as content URI.
+        Raises a ValueError when nothing has been selected
+
+        :param str title: The title is ignored on Android
+        :param initial_uri: The initial location shown in the file chooser. Must be a content URI, e.g.
+                            'content://com.android.externalstorage.documents/document/primary%3ADownload%2FTest-dir'
+        :type initial_uri: str or None
+        :param file_mime_types: The file types allowed to select. Must be MIME types, e.g.
+               ['application/pdf','application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'].
+               Currently ignored to avoid error in rubicon
+        :type file_mime_types: list[str] or None
+        :param bool multiselect: If True, then several files can be selected
+        :returns: The content URI of the chosen file or a list of content URIs when multiselect=True.
+        :rtype: str or list[str]
+        """
+        print('Invoking Intent ACTION_OPEN_DOCUMENT')
+        intent = Intent(Intent.ACTION_OPEN_DOCUMENT)
+        intent.addCategory(Intent.CATEGORY_OPENABLE)
+        intent.setType("*/*")
+        if initial_uri is not None and initial_uri != '':
+            intent.putExtra("android.provider.extra.INITIAL_URI", Uri.parse(initial_uri))
+        if file_mime_types is not None and file_mime_types != ['']:
+            # Commented out because rubicon currently does not support arrays and nothing else works with this Intent
+            # intent.putExtra(Intent.EXTRA_MIME_TYPES, file_mime_types)
+            pass
+        intent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, multiselect)
+        selected_uri = None
+        result = await self.app.invoke_intent_for_result(intent)
```

Comment:
This is awesome :D

Reply:
Yes, I also like that method. You can use it for calling any Intent you like.
In the example, I use it for calling the OI Filemanager Intents just to show how it can be used directly.